### PR TITLE
fix: MD042 empty links

### PR DESCRIPTION
**Pull request by nschonni**
Doesn't fully fix it, as it still trips on some of the text in the macros
```
files/en-us/mdn/structures/page_types/api_reference_page_template/index.md:94:3 MD042/no-empty-links No empty links [Context: "[eventname1]()"]   
files/en-us/mdn/structures/page_types/api_reference_page_template/index.md:96:28 MD042/no-empty-links No empty links [Context: "[oneventname1]()"]
files/en-us/mdn/structures/page_types/api_reference_page_template/index.md:97:3 MD042/no-empty-links No empty links [Context: "[eventname2]()"]   
files/en-us/mdn/structures/page_types/api_reference_page_template/index.md:99:28 MD042/no-empty-links No empty links [Context: "[oneventname2]()"]
files/en-us/mozilla/firefox/releases/49/index.md:73:70 MD042/no-empty-links No empty links [Context: "[@@match]()"]
files/en-us/web/api/mediakeystatusmap/index.md:35:32 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/api/webgl2renderingcontext/drawbuffers/index.md:64:40 MD042/no-empty-links No empty links [Context: "[fiuv]()"]
files/en-us/web/api/webgl2renderingcontext/index.md:90:91 MD042/no-empty-links No empty links [Context: "[v]()"]
files/en-us/web/api/webgl2renderingcontext/index.md:94:97 MD042/no-empty-links No empty links [Context: "[v]()"]
files/en-us/web/api/webgl2renderingcontext/index.md:111:88 MD042/no-empty-links No empty links [Context: "[fiuv]()"]
files/en-us/web/api/webgl2renderingcontext/index.md:143:98 MD042/no-empty-links No empty links [Context: "[if]()"]
files/en-us/web/api/webglrenderingcontext/getuniformlocation/index.md:35:64 MD042/no-empty-links No empty links [Context: "[v]()"]
files/en-us/web/api/webglrenderingcontext/getuniformlocation/index.md:39:71 MD042/no-empty-links No empty links [Context: "[fv]()"]
files/en-us/web/api/webglrenderingcontext/index.md:279:88 MD042/no-empty-links No empty links [Context: "[v]()"]
files/en-us/web/api/webglrenderingcontext/index.md:283:95 MD042/no-empty-links No empty links [Context: "[v]()"]
files/en-us/web/api/webglrenderingcontext/uniform/index.md:23:88 MD042/no-empty-links No empty links [Context: "[v]()"]
files/en-us/web/javascript/reference/global_objects/array/index.md:413:60 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/bigint64array/index.md:100:61 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/biguint64array/index.md:100:62 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/float32array/index.md:98:60 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/float64array/index.md:98:60 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/int16array/index.md:98:58 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/int32array/index.md:99:58 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/int8array/index.md:99:57 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/map/index.md:305:44 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@match/index.md:105:61 MD042/no-empty-links No empty links [Context: "[@@replace]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@match/index.md:106:60 MD042/no-empty-links No empty links [Context: "[@@search]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@match/index.md:107:59 MD042/no-empty-links No empty links [Context: "[@@split]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@replace/index.md:120:59 MD042/no-empty-links No empty links [Context: "[@@match]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@replace/index.md:121:60 MD042/no-empty-links No empty links [Context: "[@@search]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@replace/index.md:122:59 MD042/no-empty-links No empty links [Context: "[@@split]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@search/index.md:101:59 MD042/no-empty-links No empty links [Context: "[@@match]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@search/index.md:102:61 MD042/no-empty-links No empty links [Context: "[@@replace]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@search/index.md:103:59 MD042/no-empty-links No empty links [Context: "[@@split]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@split/index.md:104:59 MD042/no-empty-links No empty links [Context: "[@@match]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@split/index.md:105:61 MD042/no-empty-links No empty links [Context: "[@@replace]()"]
files/en-us/web/javascript/reference/global_objects/regexp/@@split/index.md:106:60 MD042/no-empty-links No empty links [Context: "[@@search]()"]
files/en-us/web/javascript/reference/global_objects/regexp/index.md:100:59 MD042/no-empty-links No empty links [Context: "[@@match]()"]
files/en-us/web/javascript/reference/global_objects/regexp/index.md:102:62 MD042/no-empty-links No empty links [Context: "[@@matchAll]()"]
files/en-us/web/javascript/reference/global_objects/regexp/index.md:104:61 MD042/no-empty-links No empty links [Context: "[@@replace]()"]
files/en-us/web/javascript/reference/global_objects/regexp/index.md:106:60 MD042/no-empty-links No empty links [Context: "[@@search]()"]
files/en-us/web/javascript/reference/global_objects/regexp/index.md:108:59 MD042/no-empty-links No empty links [Context: "[@@split]()"]
files/en-us/web/javascript/reference/global_objects/set/index.md:61:56 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:24:48 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:25:58 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:26:50 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:27:44 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:28:44 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:94:48 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:95:58 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:96:50 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:97:44 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/iterator/index.md:98:44 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/symbol/match/index.md:60:47 MD042/no-empty-links No empty links [Context: "[@@match]()"]
files/en-us/web/javascript/reference/global_objects/symbol/matchall/index.md:20:134 MD042/no-empty-links No empty links [Context: "[@@matchAll]()"]
files/en-us/web/javascript/reference/global_objects/symbol/matchall/index.md:51:98 MD042/no-empty-links No empty links [Context: "[@@matchAll]()"]
files/en-us/web/javascript/reference/global_objects/symbol/matchall/index.md:65:50 MD042/no-empty-links No empty links [Context: "[@@matchAll]()"]
files/en-us/web/javascript/reference/global_objects/symbol/replace/index.md:16:73 MD042/no-empty-links No empty links [Context: "[@@replace]()"]
files/en-us/web/javascript/reference/global_objects/symbol/replace/index.md:52:49 MD042/no-empty-links No empty links [Context: "[@@replace]()"]
files/en-us/web/javascript/reference/global_objects/symbol/search/index.md:16:72 MD042/no-empty-links No empty links [Context: "[@@search]()"]
files/en-us/web/javascript/reference/global_objects/symbol/search/index.md:52:48 MD042/no-empty-links No empty links [Context: "[@@search]()"]
files/en-us/web/javascript/reference/global_objects/symbol/split/index.md:16:71 MD042/no-empty-links No empty links [Context: "[@@split]()"]
files/en-us/web/javascript/reference/global_objects/symbol/split/index.md:50:47 MD042/no-empty-links No empty links [Context: "[@@split]()"]
files/en-us/web/javascript/reference/global_objects/typedarray/entries/index.md:75:26 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/typedarray/index.md:235:26 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/typedarray/keys/index.md:75:26 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/uint16array/index.md:99:59 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/uint32array/index.md:98:59 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/uint8array/index.md:99:58 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
files/en-us/web/javascript/reference/global_objects/uint8clampedarray/index.md:98:65 MD042/no-empty-links No empty links [Context: "[@@iterator]()"]
```

### CODE REVIEW

File: files/en-us/web/html/attributes/rel/index.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ The `rel` attribute has no default value. If the attribute is omitted or if none
...
 - {{htmlattrdef("dns-prefetch")}}
-  - : Relevant for the {{htmlelement('link')}} element both in the {{htmlelement('body')}} and {{htmlelement('head')}}, it tells the browser to preemptively perform DNS resolution for the target resource's origin. Useful for resources the user will likely need, it helps reduce latency and thereby improves performance when the user does access the resources as the browser preemptively performed DNS resolution for the origin of the specified resource. See [dns-prefetch](/en-US/docs/Web/Performance/dns-prefetch) described in [resource hints]().
+  - : Relevant for the {{htmlelement('link')}} element both in the {{htmlelement('body')}} and {{htmlelement('head')}}, it tells the browser to preemptively perform DNS resolution for the target resource's origin. Useful for resources the user will likely need, it helps reduce latency and thereby improves performance when the user does access the resources as the browser preemptively performed DNS resolution for the origin of the specified resource. See [dns-prefetch](/en-US/docs/Web/Performance/dns-prefetch) described in resource hints.
```
Comment (nschonni) [inline files/en-us/web/html/attributes/rel/index.md:91]: Not sure if there is a page to point to, but it wasn't obvious, so I removed the empty link

Comment (Elchi3) [reply files/en-us/web/html/attributes/rel/index.md:91]: I think this refers to the https://w3c.github.io/resource-hints/ specification.

Comment (nschonni) [inline files/en-us/web/html/attributes/rel/index.md:91]: Is this what you're suggesting?
```suggestion
  - : Relevant for the {{htmlelement('link')}} element both in the {{htmlelement('body')}} and {{htmlelement('head')}}, it tells the browser to preemptively perform DNS resolution for the target resource's origin. Useful for resources the user will likely need, it helps reduce latency and thereby improves performance when the user does access the resources as the browser preemptively performed DNS resolution for the origin of the specified resource. See [dns-prefetch](/en-US/docs/Web/Performance/dns-prefetch) described in [resource hints](https://w3c.github.io/resource-hints/).
```

Comment (Elchi3) [reply files/en-us/web/html/attributes/rel/index.md:91]: Yes!

**Review by Elchi3 — APPROVED** [review]

**Pull request merged**